Windows/mingw_defines.h

```diff
@@ -0,0 +1,234 @@
+#pragma once
```

Comment:
Is this file needed, why?

Reply:
this is needed because the headers from the dx9sdk folder aren't fully compatible with mingw.

Reply:
Again, add a comment to the code saying so, for the benefit of future readers :)